hive.yaml

```diff
@@ -0,0 +1,100 @@
+package:
+  name: hive
+  version: 4.0.0
+  epoch: 0
+  description: Apache Hive
+  copyright:
+    - license: Apache-2.0
+  dependencies:
+    runtime:
+      - bash
+      - openjdk-8-default-jvm
+
+environment:
+  contents:
+    packages:
+      - bash
+      - busybox
+      - maven
+      - openjdk-8
+  environment:
+    LANG: en_US.UTF-8
+    JAVA_HOME: /usr/lib/jvm/java-1.8-openjdk
+    HIVE_VERSION: 4.0.0
+    HADOOP_VERSION: 3.3.6
+    TEZ_VERSION: 0.10.4
+
+pipeline:
+  - uses: git-checkout
+    with:
+      repository: https://github.com/apache/hive
+      tag: rel/release-${{package.version}}
+      expected-commit: 183f8cb41d3dbed961ffd27999876468ff06690c # rel/release-4.0.0
+
+  - uses: patch
+    with:
+      # This patch covers license check for the goal org.apache.rat:apache-rat-plugin:0.13:check
+      # because the .gitconfig file for safe directories at build time should be ignored.
+      patches: rat.patch
+
+  - runs: |
+      mvn package -DskipTests -Dmaven.javadoc.skip=true -Pdist \
+        -Dhadoop.version="${HADOOP_VERSION}" \
+        -Dtez.version="${TEZ_VERSION}"
+      mkdir -p ${{targets.destdir}}/usr/share/java/hive
+      tar -xf packaging/target/apache-hive-${{package.version}}-bin.tar.gz \
+        --strip-component=1 -C ${{targets.destdir}}/usr/share/java/hive
+
+test:
+  environment:
+    contents:
+      packages:
+        - bash
+        - gpg
+        - gpg-agent
+        - gnupg-dirmngr
+        - curl
+    environment:
+      JAVA_HOME: /usr/lib/jvm/java-1.8-openjdk
+      HADOOP_VERSION: 3.3.6
+      HADOOP_HOME: /home/build/hadoop
+      HIVE_HOME: /usr/share/java/hive
+  pipeline:
+    - name: Download Hadoop
+      runs: |
+        gpg --keyserver hkps://keyserver.ubuntu.com --recv-key CD32D773FF41C3F9E74BDB7FB362E1C021854B9D
```

Comment:
I think that using `fetch` pipeline would be preferable here.

it will need updating any time HADOOP_VERSION needs updating (versus the gpg path that would accept it if it was signed), but fetch is used in more places, so we can improve that pipeline for any reason and get benefit everywhere.

then you can drop the gpg, gpg-agent, gnupg-dirmngr and curl from above.
```yaml
test:
  pipeline:
    - uses: fetch
      with:
        uri: https://dlcdn.apache.org/hadoop/common/hadoop-${{vars.hadoop-version}}/hadoop-${{vars.hadoop-version}}.tar.gz
        expected-sha256: f5195059c0d4102adaa7fff17f7b2a85df906bcb6e19948716319f9978641a04

```